website/addons/s3/static/s3NodeConfig.js

```diff
@@ -91,31 +91,45 @@ ViewModel.prototype.updateBucketList = function(){
 
 ViewModel.prototype.selectBucket = function() {
     var self = this;
+
+    // The current version of boto does not handle bucket names containing periods
+    var isValidBucket = /^(?!.*(\.\.|-\.))[^.][a-z0-9\d-]{2,61}[^.]$/;
```

Comment:
The above is just a suggestion; it may not be worth it. I think having two separate `isValidBucket` functions is fine (but I still think they should be functions, not regexes).